apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -62,28 +63,70 @@ export const collectionRouter = router({
           resourceId: !!parentFolderId ? String(parentFolderId) : null,
         })
 
-        const result = await db
-          .insertInto("Resource")
-          .values({
-            permalink,
-            siteId,
-            type: ResourceType.Collection,
-            title: collectionTitle,
-            parentId: parentFolderId ? String(parentFolderId) : null,
-            state: ResourceState.Published,
-          })
-          .returning(defaultCollectionSelect)
-          .executeTakeFirstOrThrow()
-          .catch((err) => {
-            if (get(err, "code") === PG_ERROR_CODES.uniqueViolation) {
+        const result = await db.transaction().execute(async (tx) => {
+          const user = await db
+            .selectFrom("User")
+            .where("id", "=", ctx.user.id)
+            .selectAll()
+            .executeTakeFirstOrThrow(
+              () => new TRPCError({ code: "BAD_REQUEST" }),
+            )
```

Comment:
should we move this outside the transaction so that it's non-blocking? it's read-only and doesn't seem like it has to be a all-or-nothing tx

Reply:
done